com.minres.coredsl/src/com/minres/coredsl/analysis/CoreDslAnalyzer.xtend

```diff
@@ -726,9 +726,16 @@ class CoreDslAnalyzer {
 
 				// being implicitly convertible is not good enough for alias assignments,
 				// because the alias and its source must have exactly matching bit patterns
-				if(valueType.isValid && valueType != type) {
-					ctx.acceptError("Alias must be initialized with exactly the same type it is declared as (" + type + "), but got " + valueType,
-						declarator, CoreDslPackage.Literals.DECLARATOR__TEQUALS, -1, IssueCodes.InvalidAssignmentType);
+				if(valueType.isValid && valueType != aliasType) {
+					
+					// the exception to above rule is the result of the questionable design decision to have the range
+					// access operator return a concatenated unsigned<N*bitsizeof(T)> instead of an array.
+					// it means we need to explicitly allow alias initializations where the source is a range access
+					// with the correct size and element type
+					if(!isValidRangeAlias(ctx, declarator, aliasType, initializer.value)) {
```

Comment:
I still believe this is a bad design decision and that the range operator should instead return an array.

Reply:
Duly noted.